upstream.go

```diff
@@ -1278,6 +1259,35 @@ func (uc *upstreamConn) handleMessage(msg *irc.Message) error {
 				Params:  []string{dc.nick, upstreamChannel, trailing},
 			})
 		})
+	case irc.ERR_UNKNOWNCOMMAND, irc.RPL_TRYAGAIN:
+		var command, reason string
+		if err := parseMessageParams(msg, nil, &command, &reason); err != nil {
+			return err
+		}
+
+		if command == "LIST" {
+			ok := uc.endPendingLISTs(false)
+			if !ok {
+				return fmt.Errorf("unexpected response for LIST: %q: no matching pending LIST", msg.Command)
+			}
+			uc.forEachDownstreamByID(downstreamID, func(dc *downstreamConn) {
```

Comment:
This is duplicated in the `else` branch, we can move this out of the `if` instead